cohortextractor/main.py

```diff
@@ -1,31 +1,25 @@
 import csv
-import importlib
+import importlib.util
 import inspect
-import os
-import sys
-from pathlib import Path
 
 from .backends import BACKENDS
 from .query_utils import get_column_definitions
 
 
-def main(workspace="/workspace", backend_id=None, db_url=None):
-    if not db_url:
-        db_url = os.environ["TPP_DATABASE_URL"]
-    if not backend_id:
-        backend_id = os.environ["BACKEND"]
+def main(workspace, definition, backend_id, db_url):
     backend = BACKENDS[backend_id](db_url)
-    cohort = load_cohort(workspace)
+    cohort = load_cohort(definition)
     results = extract(cohort, backend)
     write_output(results, workspace)
 
 
-def load_cohort(workspace):
-    sys.path.append(workspace)
-    study_definition = importlib.import_module("study_definition")
+def load_cohort(definition):
+    spec = importlib.util.spec_from_file_location("cohort_definition", definition)
+    cohort_definition = importlib.util.module_from_spec(spec)
+    spec.loader.exec_module(cohort_definition)
```

Comment:
This is using a fairly esoteric corner of the standard library -- could you add a comment to explain what's going on?

Reply:
As you can imagine, what's going on is code being copied from Stackoverflow. Clarifying function and comment added. I'm actually going to remove this in my next PR anyway.